include/NeoN/core/logging.hpp

```diff
@@ -0,0 +1,171 @@
+// SPDX-FileCopyrightText: 2025 NeoN authors
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <chrono>
+#include <source_location>
+#include <format>
+#include <memory>
+#include <string>
+#include <string_view>
+
+
+namespace NeoN::Logging
+{
+
+
+enum Level
+{
+    Trace,
+    Debug,
+    Info,
+    Warning,
+    Error,
+    Critical
+};
+
+enum Target
+{
+    Console,
+    File
+};
+
+/* @brief A class to represent a LogEvent
+ *
+ */
+class LogEvent
+{
+
+public:
+
+    LogEvent(std::source_location location, Level level, std::string message)
```

Comment:
Use `std::string_view` for message if possible
Avoid copying message every time

```suggestion
    LogEvent(std::source_location location, Level level, std::string_view message)
```